app/Models/Forum/Topic.php

```diff
@@ -195,6 +195,19 @@ public function nthPost($n)
         return $this->posts()->skip(intval($n) - 1)->first();
     }
 
+    public function isDoublePostBy($userId)
+    {
+        $post = $this->posts()->orderBy('post_id', 'desc')->first();
+        $postTime = $post->post_time;
+        $daysSinceLastPost = Carbon::now()->diffInDays($postTime);
+
+        if (! is_null($userId) && $post->poster_id == $userId && $daysSinceLastPost < 3) {
```

Comment:
`is_null($userId)` check is too late, it should be the first thing checked and return immediately if null.

Also I think it should always return true if `$userId` is null.

And no need to bother doing `if (condition) { return true; }`, just write it like this:

```
return $post->poster_id === ... && ...;
```

Extra: don't use `==` operator.